### PR TITLE
Add sctp library

### PR DESCRIPTION
**Pull request by adibrastegarnia**
This PR implements an initial version of SCTP library for the purpose of using it in E2T and has the following features: 

* Supports one-to-one and one-to-many SCTP interface models 
* Supports SCTP event notification and provides abstractions to determine the type of an event
   * SCTP  suggests two ways to enable the notifications - SCTP_EVENT and SCTP_EVENTS socket options.  At the moment,  in client side we support SCTP_EVENTS but SCTP_EVENT is the more generic one and it can be easily added. 

* Peeling Off an Association (for one to many interface model)

As an example (from E2T after migration): the server code will be look like this:

```
addr := &addressing.Address{
		IPAddrs: []net.IPAddr{},
		Port:    s.options.Port,
	}

	ln, err := listener.NewListener(addr,
		listener.WithInitMsg(types.InitMsg{}),
		listener.WithMode(types.OneToOne),
		listener.WithNonBlocking(false))
	if err != nil {
		return err
	}
	s.lis = ln

	go func() {
		for {
			conn, err := ln.Accept()
			if err != nil {
				log.Errorf("Failed to accept connection: %v", err)
				continue
			}

			log.Infof("Accepted connection from %s", conn.RemoteAddr())
			go handler(conn)
		}
	}()

```

and the client code for Connect will be like this:
```
addr, err := addressing.ResolveAddress(types.Sctp4, address)
	if err != nil {
		return nil, err
	}
	conn, err := sctp.DialSCTP(addr,
		sctp.WithAddressFamily(addr.AddressFamily),
		sctp.WithNonBlocking(false),
		sctp.WithMode(types.OneToOne),
		sctp.WithInitMsg(types.InitMsg{}))
	if err != nil {
		return nil, err
	}

```

Still lots of improvements should be made and unit tests also should be improved.

### CODE REVIEW

File: docs/sctp/sctp.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# SCTP Golang Library 
+
```
Comment (SeanCondon) [inline docs/sctp/sctp.md:2]: you should give a reference to the SCTP documentation, and some notes about how you chose to implement it

**Review by SeanCondon — COMMENTED** [review]
@adibrastegarnia I have no way of testing it of course but it looks good

I think it is not related to, or dependent on anything else in `onos-lib-go` - I thing it should get it's own `go.mod` and `go.sum` (inside the `sctp` folder) so that it can be reused independently.

File: docs/sctp/sctp.md (hunk 1 above)
Comment (adibrastegarnia) [reply docs/sctp/sctp.md:2]: Yeah I will add more to it.

**Comment by adibrastegarnia** [comment]
> @adibrastegarnia I have no way of testing it of course but it looks good
> 
> I think it is not related to, or dependent on anything else in `onos-lib-go` - I thing it should get it's own `go.mod` and `go.sum` (inside the `sctp` folder) so that it can be reused independently.

@SeanCondon If you have Linux, you can play with the tests in sctp_tests.go. I put // +build !darwin on top of the tests that people on Mac can run make tests. For sure it can be used independently specially since there is no good library out there.

**Comment by adibrastegarnia** [comment]
> > @adibrastegarnia I have no way of testing it of course but it looks good
> > I think it is not related to, or dependent on anything else in `onos-lib-go` - I thing it should get it's own `go.mod` and `go.sum` (inside the `sctp` folder) so that it can be reused independently.
> 
> @SeanCondon thanks for the review. If you have Linux, you can play with the tests in sctp_tests.go. I put // +build !darwin on top of the tests that people on Mac can run make tests. For sure it can be used independently specially since there is no good library out there.

FYI: I also added an example under examples folder and a Quick start @SeanCondon

File: docs/sctp/quick_start.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Then you need to import the required packages under [SCTP folder](../../pkg/sctp) in your source code files. For example:
+
+```go
+import "github.com/onosproject/onos-lib-go/pkg/sctp"
+```
+
+## A Simple SCTP Client and Server Example
+
+A simple client and a server are written as an example to show the usage of the library
```
Comment (SeanCondon) [inline docs/sctp/quick_start.md:18]: nice

File: pkg/sctp/examples/client/client.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"bytes"
+	"flag"
+	"io"
+	"strconv"
+	"time"
+
+	"log"
+
+	petname "github.com/dustinkirkland/golang-petname"
```
Comment (kuujo) [inline pkg/sctp/examples/client/client.go:26]: Is this the same library we're using in Helmit? We should just make sure we're using the same library for this so we're not adding unnecessary dependencies.

**Review by kuujo — APPROVED** [review]

File: pkg/sctp/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+// BoolToInt ...
+func BoolToInt(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+func checkEndian() binary.ByteOrder {
```
Comment (kuujo) [inline pkg/sctp/utils/utils.go:35]: This is a bad name for a function... It's not checking anything. It's getting the platform byte order.

File: pkg/sctp/examples/client/client.go (hunk 3 above)
Comment (adibrastegarnia) [reply pkg/sctp/examples/client/client.go:26]: Yeah it is the same library.

File: pkg/sctp/utils/utils.go (hunk 4 above)
Comment (adibrastegarnia) [reply pkg/sctp/utils/utils.go:35]: @kuujo what should I name it? what do you suggest?

Comment (kuujo) [reply pkg/sctp/utils/utils.go:35]: `getNativeByteOrder`

**Pull request merged**